AttributeError: 'QuerySetSequence' object has no attribute '_prefetch_related_lookups'

Any idea why ModelChoiceIterator does not work with QSS anymore ? Will look further into this anyway, apparently it broke django-autocomplete-light

```
self = <django.forms.models.ModelChoiceIterator object at 0x7f801689ced0>

    def __iter__(self):

        if self.field.empty_label is not None:

            yield ("", self.field.empty_label)

        queryset = self.queryset

        # Can't use iterator() when queryset uses prefetch_related()

>       if not queryset._prefetch_related_lookups:

E       AttributeError: 'QuerySetSequence' object has no attribute '_prefetch_related_lookups'

../.tox/base-py37-django21-sqlite/lib/python3.7/site-packages/django/forms/models.py:1137: AttributeError
```

--- a/queryset_sequence/__init__.py
+++ b/queryset_sequence/__init__.py
@@ -331,6 +331,14 @@ class QuerySetSequence(ComparatorMixin):
         if self._result_cache is None:
             self._result_cache = list(self._iterable_class(self))
 
+    @property
+    def _prefetch_related_lookups(self):
+        # for django ModelChoiceField, pending for removal
+        return any(map(
+            lambda qs: qs._prefetch_related_lookups,
+            self._querysets
+        ))
+
     # Python magic methods.
 
     def __len__(self):

--- a/queryset_sequence/__init__.py
+++ b/queryset_sequence/__init__.py
@@ -333,11 +333,17 @@ class QuerySetSequence(ComparatorMixin):
 
     @property
     def _prefetch_related_lookups(self):
-        # for django ModelChoiceField, pending for removal
-        return any(map(
-            lambda qs: qs._prefetch_related_lookups,
-            self._querysets
-        ))
+        # A hack for ModelChoiceField, which uses internal APIs from QuerySet.
+        # This should be fixed when https://code.djangoproject.com/ticket/29984
+        # is fixed.
+        #
+        # Note that this really just needs to return a truth-y value if any of
+        # the QuerySets are using prefetch_related, but this tries to keep the
+        # type sane at least.
+        result = ()
+        for qs in self._querysets:
+            result += qs._prefetch_related_lookups
+        return result
 
     # Python magic methods.
 

--- a/tests/test_querysetsequence.py
+++ b/tests/test_querysetsequence.py
@@ -340,6 +340,14 @@ class TestRelated(TestBase):
             authors = [b.author.id for b in books]
         self.assertEqual(authors, self.EXPECTED_ORDER)
 
+    def test_prefetch_related_lookups(self):
+        self.assertFalse(
+            self.all._prefetch_related_lookups
+        )
+        self.assertTrue(
+            self.all.prefetch_related('author')._prefetch_related_lookups
+        )
+
     # TODO Add a test for prefetch_related that follows multiple ForeignKeys.
 
     def test_clear_prefetch_related(self):

--- a/tests/test_querysetsequence.py
+++ b/tests/test_querysetsequence.py
@@ -3,6 +3,7 @@ from unittest import skip, skipIf
 from unittest.mock import patch
 
 import django
+from django import forms
 from django.core.exceptions import (FieldDoesNotExist,
                                     FieldError,
                                     MultipleObjectsReturned,
@@ -340,14 +341,6 @@ class TestRelated(TestBase):
             authors = [b.author.id for b in books]
         self.assertEqual(authors, self.EXPECTED_ORDER)
 
-    def test_prefetch_related_lookups(self):
-        self.assertFalse(
-            self.all._prefetch_related_lookups
-        )
-        self.assertTrue(
-            self.all.prefetch_related('author')._prefetch_related_lookups
-        )
-
     # TODO Add a test for prefetch_related that follows multiple ForeignKeys.
 
     def test_clear_prefetch_related(self):
@@ -361,6 +354,13 @@ class TestRelated(TestBase):
     def test_empty_prefetch_related(self):
         """Calling prefetch_related on an empty QuerySetSequence doesn't error."""
         self.empty.prefetch_related('author')
+
+    def test_model_choice_iterator(self):
+        """Ensure that ModelChoiceField works for QuerySetSequence."""
+        f = forms.ModelChoiceField(self.all)
+        self.assertEqual(list(f.choices), [
+            ('', '---------'),
+        ] + [(it.pk, str(it)) for it in self.all])
 
 
 class TestDeferOnly(TestBase):

